src/components/EditorPage/Metrics/Metrics.tsx

```diff
@@ -26,16 +27,11 @@ export default class Metrics extends React.PureComponent<Props, State> {
   }
 
   componentWillReceiveProps(nextProps: Props) {
-    for (let key in nextProps.metrics) {
-      const metric = key as keyof SceneMetrics
-      if (nextProps.metrics[metric] > nextProps.limits[metric]) {
-        if (!this.metricsExceeded.includes(metric)) {
-          this.metricsExceeded.push(metric)
-          this.analytics.track('Metrics exceeded', { metric })
-        }
-      } else {
-        this.metricsExceeded = this.metricsExceeded.filter(exceeded => exceeded !== metric)
-      }
+    const { metrics, limits } = nextProps
+    this.metricsExceeded = getExceededMetrics(metrics, limits)
+
+    for (const metric of this.metricsExceeded) {
+      this.analytics.track('Metrics exceeded', { metric })
```

Comment:
uh-oh, this will send the exceeded metrics to segment on every `componentWillReceiveProps`. The previous implementation would only track the event once the metric is exceeded and would not report again unless the metric goes below the limit and above again

Reply:
Oh snap, did not realised that was intended. Will fix

Reply:
@cazala Updated! let me know 😄